components/brave_wallet/browser/brave_wallet_pin_service.cc

```diff
@@ -85,14 +85,37 @@ absl::optional<mojom::WalletPinServiceErrorCode> StringToErrorCode(
   return absl::nullopt;
 }
 
-absl::optional<std::string> ExtractCID(const std::string& ipfs_url) {
-  GURL gurl = GURL(ipfs_url);
+absl::optional<std::string> ExtractIpfsUrl(const std::string& url) {
+  GURL gurl = GURL(url);
+
+  if (!gurl.is_valid()) {
+    return absl::nullopt;
+  }
+
+  if (gurl.SchemeIs(ipfs::kIPFSScheme)) {
+    return gurl.spec();
+  }
+
+  auto scope = ipfs::ExtractSourceFromGateway(gurl);
```

Comment:
nit: why is it called scope rather than like source? Is it some specific terminology used in IPFS?